core/trino-spi/src/main/java/io/trino/spi/exchange/ExchangeSource.java

```diff
@@ -48,6 +48,21 @@
      */
     void noMoreSourceHandles();
 
+    /**
+     * Called by the engine to provide information about what source task output must be included
+     * and what must be skipped.
+     * <p>
+     * This method can be called multiple times and out of order.
+     * Only a newest version (see {@link ExchangeSourceOutputSelector#getVersion()}) must be taken into account.
+     * Updates with an older version must be ignored.
+     * <p>
+     * The information provided by the {@link ExchangeSourceOutputSelector} is incremental and decisions
+     * for some partitions could be missing. The implementation is free to speculate.
+     * <p>
+     * The final selector is guaranteed to contain a decision for each source partition (see {@link ExchangeSourceOutputSelector#isFinal()}).
```

Comment:
you may clarify if decision is made for given partition in some version (returned values is either EXCLUDED or INCLUDED, not UNKNOWN), the decision cannot change in newer versions.